52.md

```diff
@@ -0,0 +1,130 @@
+NIP-52
+======
+
+Calendar Events
+---------------
+
+`draft` `optional` `author:tyiu`
+
+This specification defines calendar events that represent an occurrence at a specific moment in time or between moments in time. These calendar events are replaceable and referenceable per [NIP-33](33.md), and deletable per [NIP-09](09.md).
+
+Unlike the term `calendar event` that are specific to this NIP, the term `event` is used broadly in all the NIPs to describe any kind of Nostr event. The distinction is being made here to discern between the two terms.
+
+## Calendar Events
+
+### Format
+
+The format uses a parameterized replaceable event kind `31923` with a list of tags as described below:
+* `d` (required) universally unique identifier (UUID). Generated by the client creating the calendar event.
+* `summary` (required) name of the calendar event
+* `description` (optional) detailed description of the calendar event
+* `start` (required) inclusive start unix timestamp in seconds. Must be less than `end`, if it exists.
+* `end` (optional) exclusive end unix timestamp in seconds
+* `location` (optional) location of the calendar event. e.g. address, GPS coordinates, meeting room name, link to video call
+* `g` (optional) geohash to associate calendar event with a searchable physical location as suggested as an example by [NIP-12](12.md)
+* `p` (optional, repeated) 32-bytes hex pubkey of a participant
+* `t` (optional, repeated) hashtag to categorize calendar event
+* `r` (optional, repeated) references / links to web pages, documents, video calls, recorded videos, etc.
+
+```json
+{
+  "id": <32-bytes lowercase hex-encoded sha256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <unix timestamp in seconds>,
+  "kind": "31923",
+  "tags": [
+    ["d", "<UUID>"],
+
+    ["summary", "<summary of calendar event>"],
+    ["description", "<description of calendar event>"],
+
+    // Timestamps
+    ["start", "<unix timestamp in seconds>"],
```

Comment:
In order to support recurring calendar events, I wonder if it would make sense to switch over from unix timestamps to ISO 8601 timestamps instead.

As a reference, iCalendar formats use ISO 8601.
https://en.wikipedia.org/wiki/ICalendar